pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslDataSource.java

```diff
@@ -131,14 +125,13 @@ public PslSourceOffset getHeadOffset(TopicPath topic) {
           @Override
           public void close() {}
         };
-
-    return new PslMicroBatchReader(
-        cursorClient,
-        headOffsetReader,
-        committer,
-        subscriptionPath,
-        topicPath,
-        Objects.requireNonNull(pslDataSourceOptions.flowControlSettings()),
-        topicPartitionCount);
+    try {
+      TopicPath tp =
```

Comment:
You should probably only look this up once so you don't hit rate limits on this request, and pass the topic path into the caller of this?